include/h2o.h

```diff
@@ -1053,23 +1051,27 @@ struct st_h2o_req_t {
      */
     size_t bytes_sent;
     /**
-     * counts the number of times the request has been reprocessed (excluding delegation)
+     * the number of times the request can be reprocessed (excluding delegation)
      */
-    unsigned num_reprocessed;
+    unsigned remaining_reprocesses;
     /**
-     * counts the number of times the request has been delegated
+     * the number of times the request can be delegated
      */
-    unsigned num_delegated;
+    unsigned remaining_delegations;
 
     /**
      * environment variables
      */
     h2o_iovec_vector_t env;
 
     /**
-     * error logs
+     * error logging
      */
-    H2O_VECTOR(h2o_req_error_log_t) error_logs;
+    struct {
+        h2o_buffer_t *buf;
+        void (*cb)(h2o_req_t *req, void *data, h2o_iovec_t error);
```

Comment:
The third argument should be named `msg` or something. It's the text (or the message) that describes the error.

Reply:
Do we need both `req` and `data`?
It could be the case that we do not need `data` If we consider the callback as a method belonging to `req`. Or if we consider the callback as a free-standing callback independent to `h2o_req_t`, then `req` should not be an argument.